### ISSUE TITLE
How to specify priors to tmbstan()?

### ISSUE THREAD

**Issue by telenskyt** [kaskr/tmbstan#29]
Could you please give an example how to specify priors to `tmbstan()`? Documentation doesn't cover this. Would be also nice to have it covered on the `?tmbstan` help page.

**Comment by wlandau** [comment]
@telenskyt, I agree it would be helpful to have confirmation in the documentation.

To set priors, I think it's as simple as subtracting the log likelihood of the joint prior from the negative log likelihood in the objective function, very similar to e.g. `target += log_density_of_prior(theta)` in Stan. @kaskr, is this correct?

To check, I wrote out the simple single-observation known-variance model from [BDA3](https://sites.stat.columbia.edu/gelman/book/BDA3.pdf) chapter 2.5. The `tmbstan` samples seem to agree with the closed-form posterior.

Model:

$$
\begin{aligned}
y &\sim \text{Normal}(\theta, \sigma^2) \\
\theta &\sim \text{Normal}(\mu_0, \tau_0)
\end{aligned}
$$

The posterior $p(\theta | y)$ is normal with:

$$
\begin{aligned}
\text{E}(\theta | y) &= \frac{\frac{1}{\tau_0^2} \mu_0 + \frac{1}{\sigma^2} y}{\frac{1}{\tau_0^2} + \frac{1}{\sigma^2}} \\
\text{Var}(\theta | y) &= \left ( \frac{1}{\tau_0^2} + \frac{1}{\sigma^2} \right ) ^ {-1}
\end{aligned}
$$

which is exactly what we see with `tmbstan`:

``` r
suppressPackageStartupMessages({
  library(posterior)
  library(RTMB)
  library(tmbstan)
})

y <- 10.5
sigma <- 3.1
mu_0 <- 18.2
tau_0 <- 1.7

objective <- function(parameters) {
  getAll(parameters, warn = FALSE)
  y <- OBS(y)
  -sum(dnorm(y, theta, sigma, log = TRUE)) -
    dnorm(theta, mu_0, tau_0, log = TRUE)
}

model <- MakeADFun(objective, list(theta = 0))
mcmc <- tmbstan(model, refresh = 0)
theta <- as_draws_df(mcmc)$theta
  
# True posterior mean of theta:
((mu_0 / tau_0^2) + (y / sigma^2)) / (1 / tau_0^2 + 1 / sigma^2)
#> [1] 16.41976

# Observed posterior mean of theta:
mean(theta)
#> [1] 16.38163
  
# True posterior SD of theta:
1 / sqrt(1 / tau_0^2 + 1 / sigma^2)
#> [1] 1.490581
  
# Observed posterior SD of theta:
sd(theta)
#> [1] 1.466393
  
# These results look within MCSE:
mcse_mean(theta)
#> [1] 0.04126238
mcse_sd(theta)
#> [1] 0.02510648
```

<sup>Created on 2026-01-12 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>
